### PR TITLE
feat(api): allow passing workflow identifier as id in new v2 workflow crud

### PR DESCRIPTION
**Pull request by djabarovgeorge**
### What changed? Why was the change needed?

part of the following task
https://linear.app/novu/issue/NV-4329/add-slug-to-environment-response

### Screenshots

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: apps/api/src/app/workflows-v2/usecases/delete-workflow/delete-workflow.command.ts [hunk 1]
```diff
@@ -4,5 +4,5 @@ import { IsDefined, IsString } from 'class-validator';
 export class DeleteWorkflowCommand extends EnvironmentWithUserObjectCommand {
   @IsString()
   @IsDefined()
-  workflowId: string;
+  workflowIdOrIdentifier: string;
```
Comment (SokratisVidros) [inline apps/api/src/app/workflows-v2/usecases/delete-workflow/delete-workflow.command.ts]:
```suggestion
  workflowIdOrSlug: string;
```

File: apps/api/src/app/workflows-v2/usecases/get-workflow-by-ids/get-workflow-by-ids.command.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { EnvironmentWithUserObjectCommand } from '@novu/application-generic';
+import { IsDefined, IsString } from 'class-validator';
+
+export class GetWorkflowByIdsCommand extends EnvironmentWithUserObjectCommand {
+  @IsString()
+  @IsDefined()
+  workflowIdOrIdentifier: string;
```
Comment (SokratisVidros) [inline apps/api/src/app/workflows-v2/usecases/get-workflow-by-ids/get-workflow-by-ids.command.ts]:
```suggestion
  workflowIdOrSlug: string;
```

**Review by SokratisVidros — CHANGES_REQUESTED** [review]

File: apps/api/src/app/workflows-v2/usecases/delete-workflow/delete-workflow.command.ts (hunk 1 above)
Comment (SokratisVidros) [reply apps/api/src/app/workflows-v2/usecases/delete-workflow/delete-workflow.command.ts]: As discussed offline we will proceed with `IdentifierOrInternalId`

Comment (djabarovgeorge) [reply apps/api/src/app/workflows-v2/usecases/delete-workflow/delete-workflow.command.ts]: done

**Comment by LetItRock** [comment]
@djabarovgeorge, not missing the tests updates?

**Comment by djabarovgeorge** [comment]
> @djabarovgeorge, not missing the tests updates?

You are totally right, i forgot to mention. We will have another PR with more possible ID that the API will receive and i will add tests all the possible IDs under [the same task](https://linear.app/novu/issue/NV-4494/add-slug-parser-in-the-api-requests).

**Review by SokratisVidros — APPROVED** [review]

**Pull request merged**